### DISCUSSION TITLE
LoRaWAN how to use session

### DISCUSSION THREAD

**Discussion by StevenCellist** [jgromes/RadioLib#828] [General]
I'm trying to use an ESP32-S3 with SX1262, together with LoRaWAN from RadioLib.
On first boot, it all worked fine (properly joined, nicely sending uplinks and receiving downlinks - although I still need to look how to format downlinks such that I receive what I send from TTN).
Minor note is that it defaults to SF12 (join and uplinks) which is not great practice, but I'll at least wait for ADR implementation.

Question is: how do I resume the session after a reset / reboot / deepsleep?
Just calling `node.begin()` after a power-on reset / flash reset results in `Resuming previous session ... success!`, but any subsequent uplinks and downlinks fail with codes -12 and -1105 respectively. What should I do?

**Comment by jgromes** [comment]
-1105 is `RADIOLIB_ERR_NO_RX_WINDOW`, which is probably caused by the fact that the previous uplink failed (the library keeps track of when to open the Rx window). -12 is `RADIOLIB_ERR_INVALID_FREQUENCY`, so most likely there is something wrong with the channel selection process when resuming existing session.

You could try running the library in debug mode to see if there is further information. As a workaround, you can force the the device to always join on startup by setting `force=true` in `beginOTAA`.

**Comment by StevenCellist** [comment]
Thanks! I'll do some testing soon with debug mode enabled and see what happens. Force-rejoining will also require resetting the device on TTN so that's not really helpful on v1.1 😅

**Reply by jgromes** [reply]
I don't think that's the case, the DevNonce will not get reset by force joining, only by `node.wipe()`, so it should be possible to just join again.
